lib/DidKeyDriver.js

```diff
@@ -159,31 +179,48 @@ export class DidKeyDriver {
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
    * @param {object} options - Options hashmap.
-   * @param {LDKeyPair|object} options.keyPair - Key used to generate the DID
+   * @param {object} options.keyPair - Key used to generate the DID
    *   document (either an LDKeyPair instance containing public key material,
    *   or a "key description" plain object (such as that generated from a KMS)).
-   *
+   * @param {object} [options.keyAgreementKeyPair] -  Optional
+   *   keyAgreement key pair for generating did for keyAgreement.
    * @returns {Promise<{didDocument: object, keyPairs: Map}>}
    *   Resolves with the generated DID Document, along with the corresponding
    *   key pairs used to generate it (for storage in a KMS).
    */
-  async _keyPairToDidDocument({keyPair} = {}) {
-    const verificationKeyPair =
-      await this.verificationSuite.from({...keyPair});
+  async _keyPairToDidDocument({keyPair, keyAgreementKeyPair} = {}) {
+    let {publicKeyMultibase} = keyPair;
+    if(!publicKeyMultibase && keyPair.publicKeyBase58) {
+      publicKeyMultibase = keyPair.fingerprint();
+    }
+    // get the multibaseMultikeyHeader from the public key value
+    const multibaseMultikeyHeader = getMultibaseMultikeyHeader({
+      value: publicKeyMultibase
+    });
+    const fromMultibase = this.allowedKeyTypes.get(multibaseMultikeyHeader);
+    if(!fromMultibase) {
+      throw new Error(
+        `Unsupported "multibaseMultikeyHeader", "${multibaseMultikeyHeader}".`);
+    }
+    const verificationKeyPair = await fromMultibase({...keyPair});
```

Comment:
Just seeing this...

We should just pass `{publicKeyMultibase}` here and then simplify the `createFromMultibase` helper to only work with `{publicKeyMultibase}`.

Reply:
I've made the changes here [caf5767](https://github.com/digitalbazaar/did-method-key/pull/55/commits/caf57678def59cd5532d2f8005af10ca3aa66c7a). However, after the changes, the two tests for `ecdsa multikey` is failing with this error https://github.com/digitalbazaar/did-method-key/blob/caf57678def59cd5532d2f8005af10ca3aa66c7a/lib/helpers.js#L94-L99. The `verificationKeyPair` doesn't contain a type.  This is what it looks like: 

```
{
  id: 'did:key:zDnaeucDGfhXHoJVqot3p21RuupNJ2fZrs8Lb1GV83VnSo2jR#zDnaeucDGfhXHoJVqot3p21RuupNJ2fZrs8Lb1GV83VnSo2jR',
  controller: 'did:key:zDnaeucDGfhXHoJVqot3p21RuupNJ2fZrs8Lb1GV83VnSo2jR',
  publicKey: CryptoKey {
    type: 'public',
    extractable: true,
    algorithm: { name: 'ECDSA', namedCurve: 'P-256' },
    usages: [ 'verify' ]
  },
  publicKeyMultibase: 'zDnaeucDGfhXHoJVqot3p21RuupNJ2fZrs8Lb1GV83VnSo2jR',
  secretKeyMultibase: undefined,
  export: [AsyncFunction: exportFn],
  signer: [Function: signer],
  verifier: [Function: verifier]
}
```

This is where that keypair is coming from - https://github.com/digitalbazaar/ecdsa-multikey/blob/main/lib/index.js#L74-L89. Maybe we can add a type there if the keypair doesn't have it?